addons/project_timesheet_holidays/models/hr_employee.py

```diff
@@ -19,21 +19,22 @@ def create(self, vals):
 
     def write(self, vals):
         result = super(Employee, self).write(vals)
-        if 'active' in vals:
-            if vals.get('active'):
+        if 'active' in vals or 'resource_calendar_id' in vals:
+            has_calendar_future_holidays = 'resource_calendar_id' in vals and \
+                self.env['resource.calendar.leaves'].search([
+                    ('date_from', '>=', fields.date.today()),
+                    ('calendar_id', '=', vals['resource_calendar_id'])
+                ], limit=1)
```

Comment:
We should add a condition to be sure we will not take an approved leave for a certain employee. 😬 
Also, we should fetch the global leave whom overlaps today and not only the one in the future.
```suggestion
            today = fields.Date.today()
            has_calendar_future_holidays = 'resource_calendar_id' in vals and \
                self.env['resource.calendar.leaves'].search([
                    '|', 
                        ('date_from', '>=', today),
                        '&', 
                            ('date_from', '<', today),
                            ('date_to', '>=', today),
                    ('calendar_id', '=', vals['resource_calendar_id']),
                    ('holiday_id', '=', False),
                ], limit=1)
```